tests/build/jvm-build.go

```diff
@@ -4,50 +4,41 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"io"
-	"net/http"
-	"os"
+	"strings"
 	"time"
 
+	"github.com/devfile/library/pkg/util"
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/redhat-appstudio/jvm-build-service/pkg/apis/jvmbuildservice/v1alpha1"
 
 	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
 	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
 	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	klog "k8s.io/klog/v2"
 	"knative.dev/pkg/apis"
 )
 
-var _ = framework.JVMBuildSuiteDescribe("JVM Build Service E2E tests", Pending, Label("jvm-build"), func() {
+const (
+	testProjectGitUrl     = "https://github.com/stuartwdouglas/hacbs-test-project"
+	testProjectDevfileUrl = "https://raw.githubusercontent.com/stuartwdouglas/hacbs-test-project/main/devfile.yaml"
```

Comment:
hey @psturc - so how does the use of this test project/repo here reconcile with https://github.com/redhat-appstudio/jvm-build-service/pull/187 and the use of the specific revision @stuartwdouglas told us is needed when the jvm build service dev flow creates the pipelinerun directly

Is my assumption that since you are going off of main branch, the has component stuff will simply use the right version, and we no longer need to have that sha specific ref what you are seeing?

Aside from your answer there, I'd like to get feedback from @stuartwdouglas to make sure we are not missing anything before merging.

Reply:
I think this is ok here, as long as we are careful about how we manage this project.

The specific sha allows us to add additional features to the test project and make sure they work before all the tests start using them. 

For example, when I was adding the gradle based test I did the following:
- Create a new branch in the hacbs-test-project repo, and push a change that adds a gradle dependency
- Submit a pull request to the hacbs-jvm-orchestrator repo that updates the sha, and verify that the gradle test actually works, complete with any gradle based fixes
- Merge both pull requests

As long as we are careful to just put stuff we know works into main on hacbs-test-project then I think this is fine for this repo.

Reply:
ok thanks for the confirmation @stuartwdouglas ... fwiw I believe I agree

that said, we may want to make sure that we make sure in our "in-repo" tests @stuartwdouglas @psturc has coverage to use the sha specific test repro approach, to confirm that things are OK before we merge that into main, and then affect our regression coverage here.

I do think this lines up though @psturc with what we discussed in stand up this morning, with whether we have have both a component based tests, as well as a pipelinerun based test where we then use https://github.com/stuartwdouglas/jvm-build-service/blob/646d59b9b8d4800468981323973e3d3195d54e4c/hack/examples/run-e2e-shaded-app.yaml#L12

So I think I now want to pivot on where we landed after stand up @psturc .... we should have both a component based test and a pipeline run based test that vets changes to the test project.

Reply:
cross reference - https://github.com/redhat-appstudio/jvm-build-service/pull/161#issuecomment-1219549926

but to be clear, this PR can merge before https://github.com/redhat-appstudio/jvm-build-service/pull/161 is updated to address ^^